File: 0007_ReverseInteger.cpp
/*
注意边界值的处理，也就是res大于最大的int可表示的数时，返回0
*/

Solution {
public:
    int reverse(int x) {
        if(x<10&&x>-10)
            return x;
        long int res=0;
        while(x)
        {
            int flag=x%10;
            res=res*10+flag;
            if(res<INT_MIN || res>INT_MAX)
                return 0;
            x=x/10;


        }
        return res;
    }
};

File: python/0007_ReverseInteger.py
class Solution(object):
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        s=str(abs(x))
        s=s[::-1]
        res=int(s)
        if x<0:
            res=-res
        if res < -(2**31) or res > 2**31 - 1:
            res= 0
    
        return res
        
            
